parla/parray/core.py

```diff
@@ -1,13 +1,17 @@
 from __future__ import annotations # To support self references in type checking, must be first line
 
-import numpy
-import cupy
-
 from parla.cpu_impl import cpu
-from parla.cuda import gpu
 from parla.tasks import get_current_devices
 from parla.device import Device
 
+import numpy
+try:  # if the system has no GPU
+    import cupy
+    from parla.cuda import gpu
+except (ImportError, AttributeError):
+    cupy = numpy  # work around of cupy.ndarray
+    gpu = None
+
 class PArray:
     """Multi-dimensional array on a CPU or CUDA device.
 
```

Comment:
Could you please leave a comment that PArray only considers numpy or cupy array? This is because line 34 assumes this and therefore, lefting comments looks safe to me just in case that more array types are used.